client/verta/verta/_deployment/endpoint.py

```diff
@@ -240,8 +254,12 @@ def _update_from_dict(self, update_dict):
         else:
             raise ValueError("update strategy must be \"direct\" or \"canary\"")
 
-        run = experimentrun.ExperimentRun._get_by_id(self._conn, self._conf, id=update_dict["run_id"])
-        return self.update(run, strategy)
+        if "run_id" in update_dict:
+            model_reference = experimentrun.ExperimentRun._get_by_id(self._conn, self._conf, id=update_dict["run_id"])
+        else:
+            model_reference = RegisteredModelVersion._get_by_id(self._conn, self._conf, id=update_dict["model_version_id"])
```

Comment:
Actually actually, we can combine the arg-handling with this block so we have it all in one place. I don't know if it's a best practice, but I like it to keep code chunked. What do you think?
```suggestion
        if "run_id" in update_dict and "model_version_id" in update_dict:
            raise ValueError("cannot provide both run_id and model_version_id")
        elif "run_id" in update_dict:
            model_reference = experimentrun.ExperimentRun._get_by_id(self._conn, self._conf, id=update_dict["run_id"])
        elif "model_version_id" in update_dict:
            model_reference = RegisteredModelVersion._get_by_id(self._conn, self._conf, id=update_dict["model_version_id"])
        else:
            raise RuntimeError("must provide either model_version_id or run_id")
```

Reply:
The last suggestion looks good to me! Let me give it a try.

Reply:
Fixed.